spec/prototype_build_details_comment_action_spec.rb

```diff
@@ -0,0 +1,173 @@
+        metadata: metadata,
+        footnote: '<em>Note: Google Sign-In in not available in those builds</em>'
+      )
+
+      expect(comment).to eq <<~EXPECTED_COMMENT
+        <p>📲 You can test the changes from this Pull Request in <strong>BestApp</strong> by scanning the QR code below to install the corresponding build via App Center.</p>
+        <table>
+        <tr>
+          <td rowspan='6'><img src='https://chart.googleapis.com/chart?chs=500x500&cht=qr&chl=https%3A%2F%2Finstall.appcenter.ms%2Forgs%2FBestOrg%2Fapps%2FBestApp%2Freleases%2F8888&choe=UTF-8' width='250' height='250' /></td>
+          <td width='150px'><b>App Name</b></td><td><tt>BestApp</tt></td>
+        </tr>
+        <tr><td><b>Version:Short</b></td><td>28.2</td></tr>
+        <tr><td><b>Version:Long</b></td><td>28.2.0.108</td></tr>
+        <tr><td><b>Flavor</b></td><td>Celray</td></tr>
+        <tr><td><b>App Center Build</b></td><td><a href='https://install.appcenter.ms/orgs/BestOrg/apps/BestApp/releases/8888'>Build \#8888</a></td></tr>
+        <tr><td><b>Commit</b></td><td><tt>a1b2c3f</tt></td></tr>
+        </table>
+        <em>Note: Google Sign-In in not available in those builds</em>
+      EXPECTED_COMMENT
+    end
+
+    it 'generates a HTML table comment including the direct link if provided' do
+      metadata = {
+        'Version:Short': '28.2',
+        'Version:Long': '28.2.0.108'
+      }
+
+      comment = run_described_fastlane_action(
+        appcenter_org_name: 'BestOrg',
+        appcenter_app_name: 'BestApp',
+        appcenter_release_id: 8888,
+        download_url: 'https://bestfront.cloudfront.net/feed42/bestapp-pr1357-a1b2c3f.apk',
+        metadata: metadata
+      )
+
+      expect(comment).to eq <<~EXPECTED_COMMENT
+        <p>📲 You can test the changes from this Pull Request in <strong>BestApp</strong> by scanning the QR code below to install the corresponding build via App Center.</p>
+        <table>
+        <tr>
+          <td rowspan='6'><img src='https://chart.googleapis.com/chart?chs=500x500&cht=qr&chl=https%3A%2F%2Finstall.appcenter.ms%2Forgs%2FBestOrg%2Fapps%2FBestApp%2Freleases%2F8888&choe=UTF-8' width='250' height='250' /></td>
+          <td width='150px'><b>App Name</b></td><td><tt>BestApp</tt></td>
+        </tr>
+        <tr><td><b>Version:Short</b></td><td>28.2</td></tr>
+        <tr><td><b>Version:Long</b></td><td>28.2.0.108</td></tr>
+        <tr><td><b>Direct Link</b></td><td><a href='https://bestfront.cloudfront.net/feed42/bestapp-pr1357-a1b2c3f.apk'><tt>bestapp-pr1357-a1b2c3f.apk</tt></a></td></tr>
+        <tr><td><b>App Center Build</b></td><td><a href='https://install.appcenter.ms/orgs/BestOrg/apps/BestApp/releases/8888'>Build \#8888</a></td></tr>
+        <tr><td><b>Commit</b></td><td><tt>a1b2c3f</tt></td></tr>
+        </table>
+        <em>Automatticians: You can use our internal self-serve MC tool to give yourself access to App Center if needed.</em>
+      EXPECTED_COMMENT
```

Comment:
FYI, this is how that example looks like once rendered:<hr/>

<p>📲 You can test the changes from this Pull Request in <strong>BestApp</strong> by scanning the QR code below to install the corresponding build via App Center.</p>
<table>
<tr>
  <td rowspan='6'><img src='https://chart.googleapis.com/chart?chs=500x500&cht=qr&chl=https%3A%2F%2Finstall.appcenter.ms%2Forgs%2FBestOrg%2Fapps%2FBestApp%2Freleases%2F8888&choe=UTF-8' width='250' height='250' /></td>
  <td width='150px'><b>App Name</b></td><td><tt>BestApp</tt></td>
</tr>
<tr><td><b>Version:Short</b></td><td>28.2</td></tr>
<tr><td><b>Version:Long</b></td><td>28.2.0.108</td></tr>
<tr><td><b>Direct Link</b></td><td><a href='https://bestfront.cloudfront.net/feed42/bestapp-pr1357-a1b2c3f.apk'><tt>bestapp-pr1357-a1b2c3f.apk</tt></a></td></tr>
<tr><td><b>App Center Build</b></td><td><a href='https://install.appcenter.ms/orgs/BestOrg/apps/BestApp/releases/8888'>Build #8888</a></td></tr>
<tr><td><b>Commit</b></td><td><tt>a1b2c3f</tt></td></tr>
</table>
<em>Automatticians: You can use our internal self-serve MC tool to give yourself access to App Center if needed.</em>